### PR TITLE
[Merged by Bors] - feat: `(𝓝[<] x).NeBot` instances for `Prod`, `Pi`, `OrderDual`

### PR DESCRIPTION
**Pull request by YaelDillies**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Bases.lean [hunk 1]
```diff
@@ -54,6 +54,23 @@ More fine grained instances for `FirstCountableTopology`,
 
 open Set Filter Function Topology
 
+namespace Set
+variable {ι : Type*} {α : ι → Type*} {i : ι} {s : Set ι} {t : ∀ i, Set (α i)}
+
+lemma eval_image_pi_of_not_mem [Decidable (s.pi t).Nonempty] (hi : i ∉ s) :
+    Function.eval i '' s.pi t = if (s.pi t).Nonempty then univ else ∅ := by
```
Comment (urkud) [inline Mathlib/Topology/Bases.lean:61]: This probably belongs to `Data/Set/Prod`.

Comment (YaelDillies) [reply Mathlib/Topology/Bases.lean:61]: Whoops sorry I indeed meant to put it there

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**